### ISSUE TITLE
Installer crashes on Linux

### ISSUE THREAD

**Issue by 18107** [PULSAR-Modders/pulsar-mod-loader#52]
The installer crashes immediately on Linux without a crash log.

Steps to reproduce:

1. Download PulsarModLoaderInstaller-0.11.1.25-6159078.zip
2. Extract all contents to an empty folder
3. Open a terminal in that location and run `mono PulsarModLoaderInstaller.exe`

Result:

```
$ mono PulsarModLoaderInstaller.exe

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'PulsarModLoader.PMLConfig' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Valve.Newtonsoft.Json.JsonWriter' threw an exception. ---> System.BadImageFormatException: Could not resolve field token 0x04000358, due to: Could not load type of field 'Valve.Newtonsoft.Json.Utilities.EnumUtils+<>c:<>9__1_0' (1) due to: Could not load file or assembly 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies. assembly:/home/18107/Downloads/PML/Valve.Newtonsoft.Json.dll type:<>c member:(null)
  at Valve.Newtonsoft.Json.JsonWriter.BuildStateArray () [0x0001b] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.JsonWriter..cctor () [0x001bc] in <2c5310964c4c487abfe73911114a8b49>:0 
   --- End of inner exception stack trace ---
  at Valve.Newtonsoft.Json.Linq.JTokenWriter..ctor () [0x00000] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.Linq.JToken.FromObjectInternal (System.Object o, Valve.Newtonsoft.Json.JsonSerializer jsonSerializer) [0x00016] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.Linq.JToken.FromObject (System.Object o) [0x00006] in <2c5310964c4c487abfe73911114a8b49>:0 
  at PulsarModLoader.SaveValueManager.GetValueFor[T] (PulsarModLoader.SaveValue`1[T] saveValue, T default) [0x00067] in <53a71800fe3846058dfd108f34594256>:0 
  at PulsarModLoader.SaveValue`1[T]..ctor (System.String id, UnityEngine.TextAnchor default) [0x00018] in <53a71800fe3846058dfd108f34594256>:0 
  at PulsarModLoader.PMLConfig..cctor () [0x00000] in <53a71800fe3846058dfd108f34594256>:0 
   --- End of inner exception stack trace ---
  at PulsarInjector.Injector.Main (System.String[] args) [0x00026] in <5135491733384d65bb5de5588ed9bde3>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'PulsarModLoader.PMLConfig' threw an exception. ---> System.TypeInitializationException: The type initializer for 'Valve.Newtonsoft.Json.JsonWriter' threw an exception. ---> System.BadImageFormatException: Could not resolve field token 0x04000358, due to: Could not load type of field 'Valve.Newtonsoft.Json.Utilities.EnumUtils+<>c:<>9__1_0' (1) due to: Could not load file or assembly 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies. assembly:/home/18107/Downloads/PML/Valve.Newtonsoft.Json.dll type:<>c member:(null)
  at Valve.Newtonsoft.Json.JsonWriter.BuildStateArray () [0x0001b] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.JsonWriter..cctor () [0x001bc] in <2c5310964c4c487abfe73911114a8b49>:0 
   --- End of inner exception stack trace ---
  at Valve.Newtonsoft.Json.Linq.JTokenWriter..ctor () [0x00000] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.Linq.JToken.FromObjectInternal (System.Object o, Valve.Newtonsoft.Json.JsonSerializer jsonSerializer) [0x00016] in <2c5310964c4c487abfe73911114a8b49>:0 
  at Valve.Newtonsoft.Json.Linq.JToken.FromObject (System.Object o) [0x00006] in <2c5310964c4c487abfe73911114a8b49>:0 
  at PulsarModLoader.SaveValueManager.GetValueFor[T] (PulsarModLoader.SaveValue`1[T] saveValue, T default) [0x00067] in <53a71800fe3846058dfd108f34594256>:0 
  at PulsarModLoader.SaveValue`1[T]..ctor (System.String id, UnityEngine.TextAnchor default) [0x00018] in <53a71800fe3846058dfd108f34594256>:0 
  at PulsarModLoader.PMLConfig..cctor () [0x00000] in <53a71800fe3846058dfd108f34594256>:0 
   --- End of inner exception stack trace ---
  at PulsarInjector.Injector.Main (System.String[] args) [0x00026] in <5135491733384d65bb5de5588ed9bde3>:0
```

**Comment by BadRyuner** [comment]
Checked on Ubuntu WSL, confirming the error.
Mono comes in two packages: "mono-runtime" and "mono-complete". The first simply allows you to run NetFramework 4.X. The second allows you to run NetFramework applications that go without dependencies or reference a specific dll (System.Runtime.Serializition 3.0.0.0, hello from .NetFramework 3.0 in the NetFramework 4.X world). 
The Mono-Complete package allows you to run such applications. Just install it and the problem will disappear.
`apt install mono-complete`

@DragonFire47 , you can add this to the FAQ and close the issue

**Comment by 18107** [comment]
It looks like mono-runtime is automatically installed when switching the game to the mono branch. Would it be possible to include the extra dependencies in the download zip rather than requiring the user to install mono-complete?

It looks like only 2 extra files are needed. Downloading an entire package for 2 files seems like overkill to me.

Additional info (that probably should have been included in the first comment):
Only the 2 latest downloads crash. All previous versions run as intended.
A workaround I have been using is to run an old installer, then manually copy the latest PulsarModLoader.dll into the Managed folder.

**Comment by 18107** [comment]
The mod loader already includes 7 library files. One of which is specifically make to stop crashes on Linux (System.Windows.Forms.dll). I don't see a problem with increasing it to 9. I suspect that the files are not needed in their entirety, so a stripped down file could be used instead (like the included System.Windows.Forms.dll).

**Comment by DragonFire47** [comment]
This issue has been resolved with commit [e9a8dd6](https://github.com/PULSAR-Modders/pulsar-mod-loader/commit/e9a8dd633f30530449dc5877703d1bf08e0440e9), included in PML release 0.12.0.

The issue was related to functions used by the injector being placed in PulsarModLoader.dll, causing it to try loading it's dependencies which are not included with the release.
